### PR TITLE
Add emissions warning

### PR DESCRIPTION
**Pull request by rosypctals**
I've written some code but it's definitely wrong- I'm going to keep looking at it later!

### CODE REVIEW

File: run_echo_air.py
Comment (lkoolik) [inline run_echo_air.py:1]: Could we actually move this check to within the emissions.py init? I think that would be a little more consistent with the structure of the code. Sorry, I totally did not specify this before (or really think about it until seeing this)!

File: supporting/emissions.py [hunk 1]
```diff
@@ -434,6 +434,16 @@ def check_geo_types(self):
             
             return 
     
+    def get_max_emitting_area():
```
Comment (lkoolik) [inline supporting/emissions.py:437]: I think the issue here is you need add `self` inside the function definition (put simpler: change line 437 to `def get_max_emitting_area(self):`)

File: run_echo_air.py [hunk 2]
```diff
@@ -226,6 +226,13 @@
         conc.output_concentrations(output_region, output_dir, f_out, ca_shp_path, shape_out)
         logging.info("- [CONCENTRATION] Concentration files output into: {}.".format(output_dir))
 
+        # Find maximum emitting area
```
Comment (lkoolik) [inline run_echo_air.py:229]: I think this isn't being used anymore, right? If so, we should delete this whole chunk

File: supporting/emissions.py [hunk 3]
```diff
@@ -434,6 +437,20 @@ def check_geo_types(self):
...
+        
+        # Convert square meters to square kilometers
+        self.geometry['AREA_KM2'] = self.geometry['AREA_M2'] / 10**6
+        
+        # Get the maximum polygon area
+        max_area = self.geometry['AREA_KM2'].max()
+        
+        # Check if maximum area exceeds 2500 km^2
+        if max_area > 2500:
+            logging.info('* [EMISSIONS] Large area emissions detected ({:.2f} km2). Consider carefully if results should be used for disparity, equity, and environmental justice analyses.'.format(max_area_km2))
```
Comment (lkoolik) [inline supporting/emissions.py:452]: I think there's a little typo here and this should be:

```
 if max_area > 2500:
            logging.info('* [EMISSIONS] Large area emissions detected ({:.2f} km2). Consider carefully if results should be used for disparity, equity, and environmental justice analyses.'.format(max_area))
```

File: supporting/emissions.py [hunk 4]
```diff
@@ -434,6 +437,20 @@ def check_geo_types(self):
             
             return 
     
+    def check_polygon_area(self):
+        # Calculate polygon areas in square meters
+        self.geometry['AREA_M2'] = self.geometry['geometry'].area
```
Comment (lkoolik) [inline supporting/emissions.py:442]: I think we need to convert the CRS before this step happens. I just tested this branch with emissions from a CSV file and they are still in degrees when they reach this step. I also suspect that if emissions were input as a shapefile or feather in units of degrees that they do not get projected until after this step too. 

Can you look through the code pipeline and figure out where the emissions are projected? We want to try to minimize how many projections are happening (this slows down the code a lot). One way to do that is recognizing that the emissions are tabular, so each row only needs to be projected once (versus projected per pollutant). A second thing to be sure of is that we don't want to project too many times; if we add the projection to the ISRM grid here, let's make sure we don't also do it downstream.

**Comment by rosypctals** [comment]
There's a merge conflict because I've edited the code for the hole since I noticed the warning was being triggered for 140.0 in the CSV file where there was supposed to be no hole, so I'm assuming it's only considered a hole if it's excluding 140. If this isn't the case, please let me know! 

(aka, should the hole be >= 140 or just > 140?) 
isrm_hole_flag = sum((heights>140.0)&(heights<760.0)) > 0

**Comment by lkoolik** [comment]
> There's a merge conflict because I've edited the code for the hole since I noticed the warning was being triggered for 140.0 in the CSV file where there was supposed to be no hole, so I'm assuming it's only considered a hole if it's excluding 140. If this isn't the case, please let me know!
> 
> (aka, should the hole be >= 140 or just > 140?) isrm_hole_flag = sum((heights>140.0)&(heights<760.0)) > 0
> 
> I believe that I've restored run_echo_air.py back to normal! I've tested the code with both of the CSV files (one with hole and one without) and haven't run into errors, but besides the large shapefile, I don't have any other types of files to test to see if the large size does trigger the warning (but at least the CSV file doesn't trigger a warning?). I am currently working on seeing if the emissions are also being projected later on and making sure that we aren't projecting redudantly!

Good question! I think the hole should be just > 140 (i.e., 140 meters should not be considered "hole")

The CSV file shouldn't trigger a warning, so that is good! I think if it is triggering with the large shapefile and is not triggering with the CSV file or the mobile source 1 km gridded shapefile, then we should be good to go.

Let me know when you are ready with the emissions projection streamlining and I can review.

**Pull request merged**